### ISSUE TITLE
Issue with session timers

### ISSUE THREAD

**Issue by nbougues** [kamailio/kamailio#288]
RFC 4028 mandates that the Session-Expires header contains a refresher= parameter.

The current implementation fails against some SBCs.

Here is a small patch that solves this.

````
--- kamailio-4.2.0/modules/sst/sst_handlers.c	2014-10-16 14:46:23.000000000 +0200
+++ kamailio-4.2.0-axialys/modules/sst/sst_handlers.c	2015-04-10 09:48:36.000000000 +0200
@@ -175,13 +175,20 @@
 	sehdr->s = sst_se_buf;
 	return 0;
 }
-static inline int sst_build_se_hdr(int seval, str *sehdr)
+static inline int sst_build_se_hdr(int seval, str *sehdr, char *refresher)
 {
 	if(sehdr==NULL)
 		return -1;
 
-	sehdr->len = snprintf(sst_se_buf, SST_SE_BUF_SIZE,
-			"Session-Expires: %d\r\n", seval);
+    if(refresher==NULL) {
+        sehdr->len = snprintf(sst_se_buf, SST_SE_BUF_SIZE,
+                "Session-Expires: %d\r\n", seval);
+    }
+    else {
+        sehdr->len = snprintf(sst_se_buf, SST_SE_BUF_SIZE,
+                "Session-Expires: %d;refresher=%s\r\n", seval, refresher);
+    }
+
 	sehdr->s = sst_se_buf;
 	return 0;
 }
@@ -344,7 +351,7 @@
 		}
 		
 		info->requester = SST_PXY;
-		sst_build_se_hdr(info->interval, &msehdr);
+		sst_build_se_hdr(info->interval, &msehdr, NULL);
 		if (append_header(msg, msehdr.s)) {
 			LM_ERR("failed to append Session-Expires header to proxy "
 					"requested SST.\n");
@@ -497,6 +504,7 @@
 		struct dlg_cb_params * params) 
 {
 	struct sip_msg* msg = params->rpl;
+    char* refresher = "uac";
 
 	/*
 	 * This test to see if the message is a response sould ALWAYS be
@@ -564,7 +572,7 @@
 					 * header and forward back to the UAC and it will
 					 * deal with refreshing the session.
 					 */
-					sst_build_se_hdr(info->interval, &sehdr);
+					sst_build_se_hdr(info->interval, &sehdr, refresher);
 					if (append_header(msg, sehdr.s)) {
 						LM_ERR("failed to append Session-Expires header\n");
 						return;
```

**Comment by miconda** [comment]
Thanks.

It would be easier if you can make a pull request via github -- there are some automatic tests running for each pull request, being also easier to apply the patch.

Just be sure you format the commit message as per guidelines at:

  * http://www.kamailio.org/wiki/devel/git-commit-guidelines#commit_message

**Comment by miconda** [comment]
Fixed by @linuxmaniac with commit 5b97d81
